### PR TITLE
Add the minimal routes and config needed to get the existing huntadmin web app functionality fully working

### PR DESCRIPTION
**Pull request by obijywk**
A few of the routes have partial or missing implementations, I noted these instances with TODO comments.

### CODE REVIEW

File: src/main/java/edu/mit/puzzle/cube/core/model/User.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+        public abstract User build();
+    }
+
+    public static Builder builder() {
+        return new AutoValue_User.Builder();
+    }
+
+    @JsonProperty("username") public abstract String getUsername();
+    @JsonProperty("permissions") public abstract List<String> getPermissions();
```
Comment (tkfocht) [inline src/main/java/edu/mit/puzzle/cube/core/model/User.java]: This is going to be a problem, or at least a confusing inconsistency, if we assign permissions to roles rather than directly to users, which I think that we should due to having individual users at least on the game control team - and possibly standard and admin accounts for solving teams?

Shiro doesn't appear to have a Subject.listPermissions() or equivalent, which I totally get because that is HARD, but it means that we'll have a hard time populating this field even on a Role object.

Basically, I'm not sure what we would actually do with this in the model object, not sure we can make it actually accurate, and not sure it's in the right place.

Comment (obijywk) [inline src/main/java/edu/mit/puzzle/cube/core/model/User.java]: Yeah, that's a good insight.

The only reason I put the list of permissions there is so that the
huntadmin client can decide which UI to show (e.g. don't show the "Start
Hunt" button to a user without hunt starting permissions). There are other
ways we could address this e.g. by returning a list of roles instead of a
list of permissions. Now that I think about it, I think the client deciding
based on roles instead of permissions would be better.

I'll revise the branch later tonight to return roles instead of
permissions, if you agree that sounds better.

Comment (tkfocht) [inline src/main/java/edu/mit/puzzle/cube/core/model/User.java]: For those sorts of decisions, it might be preferable to define an /authorized route to make the decisions within the service, or what's the point of having an authorization engine. If it's one or two UI elements then it probably doesn't matter too much. As the UI decision points grow more complicated, we'd end up replicating the authorization logic in both places.

Comment (tkfocht) [inline src/main/java/edu/mit/puzzle/cube/core/model/User.java]: And I'll read through the doc during downtime today.

Comment (obijywk) [inline src/main/java/edu/mit/puzzle/cube/core/model/User.java]: On Mon, Jun 6, 2016 at 3:12 PM, TK Focht <notifications@github.com> wrote:

> In src/main/java/edu/mit/puzzle/cube/core/model/User.java
> <https://github.com/eastern-toys/Cube/pull/12#discussion_r65950163>:
>
> > +@JsonDeserialize(builder = AutoValue_User.Builder.class)
> > +public abstract class User {
> > +    @AutoValue.Builder
> > +    public static abstract class Builder {
> > +        @JsonProperty("username") public abstract Builder setUsername(String username);
> > +        @JsonProperty("permissions") public abstract Builder setPermissions(List<String> permissions);
> > +
> > +        public abstract User build();
> > +    }
> > +
> > +    public static Builder builder() {
> > +        return new AutoValue_User.Builder();
> > +    }
> > +
> > +    @JsonProperty("username") public abstract String getUsername();
> > +    @JsonProperty("permissions") public abstract List<String> getPermissions();
>
> For those sorts of decisions, it might be preferable to define an
> /authorized route to make the decisions within the service, or what's the
> point of having an authorization engine. If it's one or two UI elements
> then it probably doesn't matter too much. As the UI decision points grow
> more complicated, we'd end up replicating the authorization logic in both
> places.
>
Oh yeah, that's a good idea - I'll try implementing it that way instead and
see how it looks.

**Comment by obijywk** [comment]
I removed the User.permissions list, and added an /authorized route instead... I like it better now!

**Comment by tkfocht** [comment]
Ship it!
